skimage/morphology/extrema.py

```diff
@@ -134,27 +132,44 @@ def h_maxima(image, h, selem=None):
     #   >>> b[0] == b[1]
     #   True
     #
-    if isinstance(h, float):
-        image = image.astype(np.float)
+    if np.issubdtype(type(h), np.floating) and \
+       np.issubdtype(image.dtype, np.integer):
+        if ((h % 1) != 0):
+            warn('possible precision loss converting image to '
+                 'floating point. To silence this warning, '
+                 'ensure image and h have same data type.',
+                 stacklevel=2)
+            image = image.astype(np.float_)
+        else:
+            h = image.dtype.type(h)
+
+    if (h == 0):
+        raise ValueError("h = 0.0 is ambiguous, use local_maxima() \
+        instead?")
```

Comment:
Please use parenthesis continuation line
```suggestion
        raise ValueError("h = 0.0 is ambiguous, use local_maxima() "
                         "instead?")
```